pkg/services/rpcsrv/subscription_test.go

```diff
@@ -625,3 +625,60 @@ func TestSubscriptionOverflow(t *testing.T) {
 	finishedFlag.CompareAndSwap(false, true)
 	c.Close()
 }
+
+func TestFilteredIsValidSubscriptions(t *testing.T) {
+	priv0 := testchain.PrivateKeyByID(0)
+	var goodSender = priv0.GetScriptHash()
+
+	var cases = map[string]struct {
+		params string
+	}{
+		"tx wrong sender": {
+			params: `["transaction_added", {"unknown_sender":"` + goodSender.StringLE() + `"}]`,
+		},
+		"notification with long name": {
+			params: `["notification_from_execution", {"name":"notification_from_execution_with_long_name"}]`,
+		},
+		"execution not valid vm state": {
+			params: `["transaction_executed", {"state":"NOTHALT"}]`,
+		},
+	}
+	var s string
+	for name, this := range cases {
+		t.Run(name, func(t *testing.T) {
+			chain, rpcSrv, c, respMsgs, finishedFlag := initCleanServerAndWSClient(t)
+
+			defer chain.Close()
+			defer rpcSrv.Shutdown()
+
+			blockSubID := callSubscribe(t, c, respMsgs, `["block_added"]`)
+
+			resp := callWSGetRaw(t, c, fmt.Sprintf(`{"jsonrpc": "2.0","method": "subscribe","params": %s,"id": 1}`, this.params), respMsgs)
+			require.NotNil(t, resp.Error)
+			require.Nil(t, resp.Result)
+			require.Error(t, json.Unmarshal(resp.Result, &s))
+
+			var lastBlock uint32
+			for _, b := range getTestBlocks(t) {
+				require.NoError(t, chain.AddBlock(b))
+				lastBlock = b.Index
+			}
+
+			for {
+				resp := getNotification(t, respMsgs)
+				rmap := resp.Payload[0].(map[string]any)
+				if resp.Event == neorpc.BlockEventID {
+					index := rmap["index"].(float64)
+					if uint32(index) == lastBlock {
+						break
+					}
+					continue
+				}
+			}
+
+			callUnsubscribe(t, c, respMsgs, blockSubID)
+			finishedFlag.CompareAndSwap(false, true)
```

Comment:
You don't need this code, it's just unnecessary. The only thing you're checking here if failed "subscribe" request due to invalid filter.